xonsh/prompt/gitstatus.py

```diff
@@ -17,11 +17,27 @@
 
 def _check_output(*args, **kwargs):
     kwargs.update(dict(env=builtins.__xonsh_env__.detype(),
+                       stdout=subprocess.PIPE,
                        stderr=subprocess.DEVNULL,
-                       timeout=builtins.__xonsh_env__['VC_BRANCH_TIMEOUT'],
                        universal_newlines=True
                        ))
-    return subprocess.check_output(*args, **kwargs)
+    timeout = builtins.__xonsh_env__['VC_BRANCH_TIMEOUT']
+    # See https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate
+    with subprocess.Popen(*args, **kwargs) as proc:
+        try:
+            out, err = proc.communicate(timeout=timeout)
+            if proc.returncode != 0:
+                raise subprocess.CalledProcessError(proc.returncode, proc.args, output=out, stderr=err)  # note err will always be empty as we redirect stderr to DEVNULL abvoe
+            return out
+        except subprocess.TimeoutExpired:
+            # We use `.terminate()` (SIGTERM) instead of `.kill()` (SIGKILL) here
+            # because otherwise we guarantee that a `.git/index.lock` file will be
+            # left over, and subsequent git operations will fail.
+            # We don't want that.
+            # As a result, we must rely on git to exit properly on SIGTERM.
+            proc.terminate()
+            proc.wait()  # we ignore what git says after we sent it SIGTERM
```

Comment:
Do we need to wait on the process here? Can we just assume it'll clean itself up?